### DISCUSSION TITLE
`addVariant` doesn't work for named groups

### DISCUSSION THREAD

**Discussion by bengry** [tailwindlabs/tailwindcss#13464] [Help] [answered]
I'm trying to add a variant on a group, but it doesn't seem to work - only if the group is unnamed (i.e. `.group`, not `.group/item` for example), the plugin works as expected.

I'm using `addVariant`, but also tried with `matchVariant` and it didn't seem to work.

Playground: https://play.tailwindcss.com/ecsY8X6ox9

**Comment by wongjn** [comment]
You'd need to follow [how core does it](https://github.com/tailwindlabs/tailwindcss/blob/f1f419a9ecfcd00a2001ee96ab252739fca47564/src/corePlugins.js#L156-L159) like https://play.tailwindcss.com/Qyhp3qrvMr?file=config.

**Reply by bengry** [reply]
@wongjn thanks. this seems to break other attributes used on `group` though - see here: https://play.tailwindcss.com/V93agacyzP


Also, shouldn't this be more specific, but work the same?: https://play.tailwindcss.com/3KwEgErIPF?file=config
It doesn't, but I'm not sure why.

**Reply by wongjn** [reply]
You could use the built-in `group-data-[]` variant: https://play.tailwindcss.com/6s6vFbSBFe

**Reply by bengry** [reply]
@wongjn of course, but the whole point is to provide a "shortcut" for that common use-case, in my app.

**Reply by wongjn** [reply, marked as answer]
Use a different `name` for the first parameter of the `matchVariant()` call, with a `DEFAULT` value key so that the `name` matches by itself: https://play.tailwindcss.com/BiqH28z4Lq?file=config.
